### ISSUE TITLE
About first Int32 value in assets file header.

### ISSUE THREAD

**Issue by 0x900x90** [DerPopo/UABE#107]
It's not critical problem, but i think you have to check.

Looks like UABE counts it wrong. It gets decreased by 1 even if i just reimported unchanged exported data.

Screenshot.
https://s28.postimg.org/swnecrr4b/12wewqkiku8.png?dl=1

Only this byte differs.

Also there is some issue with deserializing PlayerSettings in globalgamemanagers. I have to change in manually. Something changed inside this block in 5.4.2.p3. I think PlayerSettings have the same format as in 5.5.0.

And another screenshot. Maybe it's only XP related issue.
http://s29.postimg.org/ob9iogczb/13123.png

**Comment by DerPopo** [comment]
Thanks for the report!
I unfortunately wasn't able to reproduce the incorrect metadata size (stored in the first value), at least in a 5.4.2f2 game. I'll keep looking for more samples that might produce the bug.

PlayerSettings regularly changes, I'd have to create a lot more type databases for it to show it correctly in all Unity releases. What I could try to do is to compute and compare the type hashes with those stored in every Unity 5 .assets file. So far, I only have all the information required to create such a hash in the 5.5.0 type database but maybe I'll add these fields to the previous ones.

The export name issue was caused by a memory copy with only half the bytes needed for the whole name (of course also missing the null terminator causing those symbols at the end).
Also fixed the other bug in the screenshot, it happens when the file table in globalgamemanagers refers to another .assets file that wasn't loaded. I just forgot to check the bool I set to determine if the .assets file is actually loaded.
